### PR TITLE
LPS-148658 Use `Util.openAlertModal` on JS files

### PR DESCRIPTION
**Pull request by diegonvs**
> Depends on: https://github.com/liferay-frontend/liferay-portal/pull/1990, https://github.com/liferay-frontend/liferay-portal/pull/2072

#### What's this PR do?

Replace existing window.alert usages on JavaScript files with the new `Util.openAlertModal`

**These changes are under `Liferay.__FF__.enableCustomDialogs` feature flag and it should be enabled by product teams gradually and they should prioritize it** /cc @dsanz @markocikos   

##### For -sample modules:

0. Fetch my changes
1. Enable the feature flag by setting `feature.flag.enableCustomDialogs=true` on portal-ext.properties file
2. Deploy frontend-taglib-clay-sample module
3. Drop clay sample module using page editor
4. Go to Management Toolbar tab
5. On Display Context section, click on Info button or check the checkbox of the management toolbar

#### Screenshots (if appropriate)

Peek 2022-03-15 18-12

#### What gif best describes this PR or how it makes you feel?

run

#### Tasks:

- [x] Migrate on -sample modules
- [x] Migrate site-initalizer-raylife usages
- [x] Migrate osb-site-initializer-liferay-online usages
- [x] Migrate test related things

### CODE REVIEW

**Comment by diegonvs** [comment]
Rebased with current https://github.com/liferay-frontend/liferay-portal/pull/1990 and updated on Liferay Online module

**Comment by diegonvs** [comment]
What is left: 

Migrate use cases of 

```
portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/dependencies/collections-HTMLCSSJSJSON-replace.zip/collection-name/fragments/fragment-name/index.js:11:			alert('Here we go');
portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/dependencies/fragment_button.js:11:			alert('Form submitted');
```

**Comment by diegonvs** [comment]
This PR was on top of today's (8 Apr) brianchandotcom fork. It was necessary because I needed some changes that aren't synced yet.

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
> This PR was on top of today's (8 Apr) brianchandotcom fork. It was necessary because I needed some changes that aren't synced yet like the Liferay.**FF**.enableCustomDialogs feature flag :) .

@diegonvs I thought our sync script stopped working :slightly_smiling_face:  Please rebase and force push. This is making reviewing difficult and it may be causing CI failures.

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js [hunk 1]
```diff
@@ -25,7 +25,14 @@ export default function propsTransformer({
 		},
 		onActionDropdownItemClick({action, itemData}) {
 			if (action.data.id === 'sampleMessage') {
-				alert(`${greeting} ${itemData.title}!`);
+				if (Liferay.__FF__.enableCustomDialogs) {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js:28]: (optional) It is so awkward to see a verb for a boolean variable :disappointed: Verbs should be used for functions/methods, `function enableCustomDialogs()`. Can we rename this to `customDialogsEnabled`?

**Review by markocikos — COMMENTED** [review]
@diegonvs Looks good. Couple of minor optional nits in comments, only the import one is important to change.

File: modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/js/ClaySampleManagementToolbarPropsTransformer.js [hunk 2]
```diff
@@ -16,22 +16,60 @@ export default function propsTransformer(props) {
...
 		onCheckboxChange: () => {
-			alert('Select all checkbox changed');
+			if (Liferay.__FF__.enableCustomDialogs) {
+				Liferay.Util.openAlertModal({
+					message: 'Select all checkbox changed',
+				});
+			}
+			else {
+				alert('Select all checkbox changed');
+			}
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/resources/META-INF/resources/js/ClaySampleManagementToolbarPropsTransformer.js:34]: Yes, I identified this opportunity when touching JSP files yesteday. I'll change this :)

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js (hunk 1 above)
Comment (diegonvs) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js:28]: Can we do this? I can easily switch then :)

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js [hunk 3]
```diff
@@ -25,7 +25,14 @@ export default function propsTransformer({
...
+					Liferay.Util.openAlertModal({
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js:29]: Yes, yes! Thanks for pointing up!

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/SampleFDSPropsTransformer.js:28]: I'd give this 80% chance to of pass Brian :smile:

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Review by markocikos — APPROVED** [review]
LGTM!

CI failures look unrelated on a glance, I'll try forwarding it.

**Comment by markocikos** [comment]
ci:forward

**Comment by diegonvs** [comment]
ci:forward

**Comment by diegonvs** [comment]
When running

1. `ant format-source-all` on `portal-impl/` folder (A dinossaur friend told me few years ago that CI runs SF from here) returns nothing.

2. `gw formatSource` on each module. I doesn't throw any errors

3. `ant format-source-current-branch` on `portal-impl/` folder throws any error

4. `ant format-source-current-branch` on `portal-impl/` folder throws any error

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
Reviewed. Test failures are unrelated. :heavy_check_mark: 

[**Testray Comparison**](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=1664590103&testrayRunIdA=1666520407&statusB=2)
- Forms tests failed prior to running test 
- [ObjectPortlet#CanSearchPartialOnRelationshipField](https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=11/44769//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_ObjectPortlet_CanSearchPartialOnRelationshipField_) has been [flaky](https://testray.liferay.com/home/-/testray/case_results/1666527364/history?tab=result&hideFilterPin=true)

Will manually forward

**Comment by diegonvs** [comment]
Thanks John :)

**Comment by john-co** [comment]
forwarded to https://github.com/brianchandotcom/liferay-portal/pull/116136

**Pull request closed without merging**